packages/emotion/src/InstUISettingsProvider/index.tsx

```diff
@@ -136,12 +150,25 @@ function InstUISettingsProvider({
       </ThemeProvider>
     </DeterministicIdContextProvider>
   )
+
+  if (dir) {
+    const Element = getElementType(InstUISettingsProvider, { as })
+    providers = <Element dir={finalDir}>{providers}</Element>
+  } else if (as && process.env.NODE_ENV !== 'production') {
+    console.warn(
+      "The 'as' property should be used in conjunction with the 'dir' property!"
```

Comment:
It might be overkill but we could write a test for this warning too.

Reply:
You are correct, I added a test case for this.